projects/kit/components/carousel/carousel-draggable.directive.ts

```diff
@@ -0,0 +1,74 @@
+import {DOCUMENT} from '@angular/common';
+import {
+    Directive,
+    ElementRef,
+    HostBinding,
+    HostListener,
+    Inject,
+    Input,
+} from '@angular/core';
+import {INTERSECTION_ROOT} from '@ng-web-apis/intersection-observer';
+import {
+    clamp,
+    TUI_IS_MOBILE,
+    tuiDefaultProp,
+    TuiPanService,
+    typedFromEvent,
+} from '@taiga-ui/cdk';
+import {EMPTY,Observable} from 'rxjs';
+import {delay, endWith, filter, map, repeat, scan, takeUntil} from 'rxjs/operators';
+
+// @dynamic
+@Directive({
+    selector: 'tui-carousel[draggable]',
+    providers: [TuiPanService],
+})
+export class TuiCarouselDraggableDirective extends Observable<string | null> {
+    private readonly transform$ = this.isMobile
+        ? EMPTY
+        : this.pan$.pipe(
+              filter(() => this.draggable),
+              scan((acc, [cur]) => acc + cur, 0),
+              takeUntil(typedFromEvent(this.documentRef, 'mouseup').pipe(delay(0))),
+              map(x => `translateX(${clamp(x, this.min, this.max)}px)`),
+              endWith(null),
+              repeat(),
+          );
+
+    private min = -Infinity;
+    private max = Infinity;
+
+    @Input()
+    @HostBinding('class._draggable')
+    @tuiDefaultProp()
+    draggable = true;
+
+    @HostBinding('class._transitioned')
+    transitioned = false;
+
+    constructor(
+        @Inject(DOCUMENT) private readonly documentRef: Document,
+        @Inject(TUI_IS_MOBILE) private readonly isMobile: boolean,
```

Comment:
do we also have to do like here?
https://github.com/TinkoffCreditSystems/taiga-ui/pull/1055/files?